### ISSUE TITLE
How to set a fixed learning rate for YOLOV8？

### ISSUE THREAD

**Issue by Vayne0227** [ultralytics/ultralytics#16098]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

How to set a fixed learning rate for YOLOV8? How to adopt a smaller learning rate after a specified epoch? For example, the number of epochs I want to train is 30, with a learning rate of 0.0003 for the first 20 epochs and 0.00003 for the next 10 epochs

**Comment by Alarmod** [comment]
_lrf_ set to 1 and _warmup_epochs_ set to 0?

**Comment by Vayne0227** [comment]
> _lrf_ set to 1 and _warmup_epochs_ set to 0?

But how to reduce the learning rate next

**Comment by Alarmod** [comment]
if _lrf_ set to 1, then learning rate always fixed as _lr0_

**Comment by Vayne0227** [comment]
> if _lrf_ set to 1, then learning rate always fixed as _lr0_

Yes, so I want to know how to reduce the learning rate after a certain number of training sessions

**Comment by Alarmod** [comment]
For examlpe, _lr0_ set to 0.01 and _lrf_ set to 0.1, so speed will be reduced from 0.01 to 0.001 to the end of training

**Comment by Vayne0227** [comment]
> For examlpe, _lr0_ set to 0.01 and _lrf_ set to 0.1, so speed will be reduced from 0.01 to 0.001 to the end of training

Yes, but I want to know how to train for 30 epochs, with a learning rate of 0.0003 for the first 20 epochs and 0.00003 for the last 10 epochs. Instead of gradually diminishing.

**Comment by glenn-jocher** [comment]
To achieve a fixed learning rate for specific epochs, you can manually adjust the learning rate in your training loop by modifying the optimizer's learning rate after the desired number of epochs. This requires custom code to update the learning rate scheduler.
